web/pdf_viewer.js

```diff
@@ -165,12 +164,21 @@ var PDFViewer = (function pdfViewer() {
         this._currentPageNumber = val;
         return;
       }
+      this._setCurrentPageNumber(val, false);
+      // The intent can be to just reset a scroll position and/or scale.
+      this._resetCurrentPageView();
+    },
 
+    _setCurrentPageNumber:
+        function pdfViewer_setCurrentPageNumber(val, updateInProgress) {
```

Comment:
I'm wondering if we actually still need the `updateInProgress` parameter here, and in the events?
Previously we called `scrollPageIntoView` from the event handler in `app.js` (or `viewer.js` as it were), which seemed bad for e.g. the viewer components. In PR #6140, I removed some dead code, and also refactored this. With the changes in this patch, it seems to me at least, that `updateInProgress` is now pointless.
(Obviously there might be someone relying on it, but given that we've not really made any promises about the stability of the viewer components, I think we can remove it.)